jina/executors/indexers/vector.py

```diff
@@ -263,8 +283,7 @@ def query(self, keys: 'np.ndarray', top_k: int, *args, **kwargs) -> Tuple['np.nd
         else:
             raise NotImplementedError(f'{self.metric} is not implemented')
 
-        idx = dist.argsort(axis=1)[:, :top_k]
-        dist = np.take_along_axis(dist, idx, axis=1)
+        idx, dist = self._find_smallest_distances(dist, top_k)
```

Comment:
like it, but I would call `_get_sorted_top_k`?

Reply:
fine for me :)